src/com/linkedin/parseq/internal/SerialExecutor.java

```diff
@@ -85,6 +97,10 @@ public void run() {
       try {
         runnable.run();
       } finally {
+        if (_pendingCount.get() == 1) {
+          _active = false;
+          _activityHandler.deactivated();
+        }
```

Comment:
This is expected and thus thread-safe. If situation you described happens there will be the following sequence called:
1) deactivated()
2) activated()
I'll put more comments to explain what is going on here.